src/basic/io-util.c

```diff
@@ -306,3 +306,18 @@ ssize_t sparse_write(int fd, const void *p, size_t sz, size_t run_length) {
 
         return q - (const uint8_t*) p;
 }
+
+_noreturn_ void report_errno_and_exit(int errno_fd, int r) {
```

Comment:
I'd like to name `r` -> `error` or so, and rename `rr` -> `r` below to follow our usual coding style.